pkg/tls/renewer.go

```diff
@@ -168,6 +188,13 @@ func (c *certRenewer) ValidateCert(ctx context.Context) (bool, error) {
 	return validateCert(time.Now(), cert, caCerts...), nil
 }
 
+func (c *certRenewer) verifyCertChain(ctx context.Context, rootCert, cert *x509.Certificate) error {
+	roots := x509.NewCertPool()
+	roots.AppendCertsFromPEM(certificateToPem(rootCert))
+	_, err := cert.Verify(x509.VerifyOptions{Roots: roots})
+	return err
+}
+
```

Comment:
There's a `ValidateCert` func, why not use it ?

Reply:
I didn't check that func! Updated.